src/common/config/index.js

```diff
@@ -324,6 +327,35 @@ export default class Config extends EventEmitter {
     return newTeams;
   }
 
+  /**
+   * Apply a default sort order to the team list, if no order is specified.
+   * @param {array} teams to sort
+   */
+  sortUnorderedTeams(teams) {
+    // Make a best pass at interpreting sort order. If an order is not specified, assume it is 0.
+    //
+    const newTeams = teams.sort((x, y) => {
+      if (x.order == null) {
+        x.order = 0;
+      }
+      if (y.order == null) {
+        y.order = 0;
+      }
+
+      // once we ensured `order` exists, we can sort numerically
+      return x.order - y.order;
+    });
+
+    // Now re-number all items from 0 to (max), ensuring user's sort order is preserved. The
+    // new tabbed interface requires an item with order:0 in order to raise the first tab.
+    //
+    newTeams.forEach((team, i) => {
+      team.order = i;
```

Comment:
Is the order here being saved back to `config.json`? Or does that not happen until the order is changed by dragging tabs?

Reply:
I'm not sure (I've only understood the code enough to fix the bug.)  How can I test that?

Reply:
Open up `config.json` in a text editor, then open Mattermost and see if the order changes on load. I assume it would only happen if there were new GPO teams it had to account for.

Reply:
Done - Just confirmed that the config file is not updated until the user changes it (add/remove server, move tabs around, etc).

Reply:
Okay cool, can you also confirm that when moving a tab and then restarting the app the tabs appear in the same order as they were before restarting?

Reply:
Yep, I tested that when I ran through the previous test. It remembered the user's desired tab sort order.

Reply:
Great, thanks :)